cocos2d/CCLabelTTF.m

```diff
@@ -479,7 +477,7 @@ - (CCTexture*) createTextureWithAttributedString:(NSAttributedString*)attributed
     
     CGSize originalDimensions = _dimensions;
   
-    CGFloat scale = [CCDirector sharedDirector].contentScaleFactor;
+    CGFloat scale = __ccContentScaleFactor;
```

Comment:
i would change this to  `CGFloat scale = self.director.contentScaleFactor` as you have on line 706